### PR TITLE
feat: namespace for prefix path, direct navigate if exited

### PR DESCRIPTION
**Pull request by Ariussssss**
1. Support namespace with params, like downloading as `freesteph/elescope`  instead of `elescope`.
2. Open source directory if exited.

Nice tool, thanks 🤩

### CODE REVIEW

File: elescope.el [hunk 1]
```diff
@@ -45,6 +45,14 @@ Defaults to 1 which makes all clones shallow clones."
   :group 'eslescope
   :type 'integer)
 
+
+(defcustom elescope-use-namespace nil
+  "Use project username as namespace for path.
+
+Default as nil"
+  :group 'byte-elescope
+  :type 'integer)
```
Comment (freesteph) [inline elescope.el]: Should probably be a boolean? Also `byte-elescope`? Wouldn't this be `elescope`?

File: elescope.el [hunk 2]
```diff
@@ -100,16 +108,18 @@ by `run-at-time'."
               (not (seq-contains path ?/))
               (equal path (alist-get 'no-results elescope--strings)))
     (let* ((url (format "https://github.com/%s" path))
-           (name (cadr (split-string path "/")))
+	   (name (if elescope-use-namespace path (cadr (split-string path "/"))))
```
Comment (freesteph) [inline elescope.el]: whitespace

File: elescope.el [hunk 3]
```diff
@@ -100,16 +108,18 @@ by `run-at-time'."
...
            (destination (expand-file-name name elescope-root-folder))
            (command (format
                      "git clone --depth=%s %s %s"
                      elescope-clone-depth
                      url
                      destination)))
-      (if (eql 0 (shell-command command))
-          (find-file destination)
-        (user-error "Something went wrong whilst cloning the project")))))
+      (if (file-directory-p destination)
```
Comment (freesteph) [inline elescope.el:126]: I believe the destination will always be a directory, do we need these lines?

File: elescope.el [hunk 4]
```diff
@@ -45,6 +45,14 @@ Defaults to 1 which makes all clones shallow clones."
   :group 'eslescope
   :type 'integer)
 
+
+(defcustom elescope-use-namespace nil
+  "Use project username as namespace for path.
+
+Default as nil"
```
Comment (freesteph) [inline elescope.el]: Could we have an example in the description? It took me a bit of time to understand the functionality, but now I'm all aboard. Maybe something like

```elisp
(defcustom elescope-use-full-path nil
"Use the full project path for the resulting clone.

If non-nil, use the full project path including
username/organisation to clone: cloning 'john/foo' and 'john/bar'
results in two 'foo' and 'bar' clones inside a parent 'john'
folder, as opposed to the default, flat hierarchy of 'foo' and
'bar'."
  :group 'elescope
  :type 'boolean)
```

Feel free to use your own style, as long as simple minds like me can understand it.

**Review by freesteph — COMMENTED** [review]
Thanks a lot! I appreciate the contribution and have added a couple comments I'd like you to amend. Happy to merge soon after.

Thanks again!

File: elescope.el (hunk 3 above)
Comment (Ariussssss) [reply elescope.el:126]: When the project has not be downloaded before and the command goes error, directory won't be existed.
And as clone the same project twice, just navigate to the destination directory without run shell-command.

File: elescope.el (hunk 1 above)
Comment (Ariussssss) [reply elescope.el]: Mistake🙈

File: elescope.el (hunk 2 above)
Comment (Ariussssss) [reply elescope.el]: I don't get it.🙈

File: elescope.el (hunk 4 above)
Comment (Ariussssss) [reply elescope.el]: I thought that the `namespace` might make it easy to recognize what I mean. And my English just poor enough🤣. Already replace it.

**Review by freesteph — APPROVED** [review]
Awesome, thank you.

**Pull request merged**